packages/ui-kit/src/components/markdown.js

```diff
@@ -111,6 +75,21 @@ const Dl = styled.dl`
 const Dt = styled.dt`
   color: ${colors.light.textSecondary};
 `;
+
+/*
+The "container" styles have to be applied to containers that render markdown inside a surrounding
+visual box like a blockquote, notification box, card, or Subtitle.
+Heading margins are not set here because headings can and should not be used inside such containers.
+*/
+const containerStyles = () => css`
+  > * + * {
+    margin-top: ${dimensions.spacings.m};
+  }
+  > * + ${Ul}, > * + ${Ol}, > * + ${Dl} {
+    margin-top: ${dimensions.spacings.s};
+  }
+`;
+
```

Comment:
"containerStyles" had to be moved down here to be able to have access to the Ul, Ol and Dl variables for nested lists in containers.